### PR TITLE
Task_2: validate user input

### PR DESCRIPTION
**Pull request by Pasichnichenkoilya**

### CODE REVIEW

File: lab2/Pasichnichenko_Task2.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    try:
+        user_input = float(input(message))
+        if isinstance(user_input, float):
+            return user_input
+    except:
+        print('It must be a number')
+        return get_float(message)
+
+
+x = get_float('Enter x: ')
```
Comment (vloooo) [inline lab2/Pasichnichenko_Task2.py:16]: "Enter {}: " is a common part for a few cases.
try to decrease duplications

File: lab2/Pasichnichenko_Task2.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+
+def get_float(message):
+    try:
+        user_input = float(input(message))
+        if isinstance(user_input, float):
+            return user_input
+    except:
+        print('It must be a number')
+        return get_float(message)
```
Comment (vloooo) [inline lab2/Pasichnichenko_Task2.py:13]: move return out of except block

File: lab2/Pasichnichenko_Task2.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# Multi-paradigm programming languages Lab2 Illia Pasichnichenko IKM-221a
+print('Multi-paradigm programming languages Lab2')
+print('Illia Pasichnichenko IKM-221a')
+
+
+def get_float(message):
+    try:
+        user_input = float(input(message))
+        if isinstance(user_input, float):
```
Comment (vloooo) [inline lab2/Pasichnichenko_Task2.py:9]: It's great that you tried checking by isinstance().
but from logic perspective it's meaningless. 
at the previous line you have casting to float.
So, there could be only 2 options: 
1. the program will fail with exception and the **if** will not be executed
2. you receive the value of float type. so, you don't need to check

File: lab2/Pasichnichenko_Task2.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    except:
+        print('It must be a number')
+        return get_float(message)
+
+
+x = get_float('Enter x: ')
+y = get_float('Enter y: ')
+z = get_float('Enter z: ')
+
+result = (11 + 2 * x + 4.1) / (12.4 - y) + z
```
Comment (vloooo) [inline lab2/Pasichnichenko_Task2.py:20]: I suspect you need one more validation for **y**.
could it be **12.4**?

File: lab2/Pasichnichenko_Task2.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
```
Comment (vloooo) [inline lab2/Pasichnichenko_Task2.py:21]: I would advise you to delete a blank line between **result** calculation and displaying the value.
these lines in may head represent the single piece of logic. So, better to have them together.

File: lab2/Pasichnichenko_Task2.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
+# Multi-paradigm programming languages Lab2 Illia Pasichnichenko IKM-221a
```
Comment (vloooo) [inline lab2/Pasichnichenko_Task2.py:1]: impressive solution

Comment (Pasichnichenkoilya) [reply lab2/Pasichnichenko_Task2.py:1]: Thanks, glad to hear that

File: lab2/Pasichnichenko_Task2.py [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+
+def validate(value, condition_lambda, error_msg):
+    while condition_lambda(value):
+        value = get_float(error_msg)
+    return value
+
+
+x = get_float('x')
+y = validate(get_float('y'), lambda _: _ == 12.4, 'the value cannot be equal to 12.4, enter again: ')
```
Comment (vloooo) [inline lab2/Pasichnichenko_Task2.py:22]: let's discuss this part. please ping me when it would be convenient for you.

Comment (Pasichnichenkoilya) [reply lab2/Pasichnichenko_Task2.py:22]: I'm free right now

**Review by vloooo — APPROVED** [review]

**Pull request merged**